integrations/googlecalendar/src/actions/sync/event-create.ts

```diff
@@ -0,0 +1,50 @@
+import { getClient } from 'src/client'
+import { mapEvent } from 'src/misc/map-event'
+import { parseError } from 'src/misc/utils'
+import * as bp from '.botpress'
+
+type Merge<T, U> = Omit<T, keyof U> & U
+
+type EventCreateInput = bp.actions.eventCreate.input.Input
+type EventCreateProps = Merge<
+  bp.ActionProps['eventCreate'],
+  {
+    input: Merge<
+      EventCreateInput,
+      {
+        item: Omit<EventCreateInput['item'], 'id'>
+      }
+    >
+  }
+>
+
+export const eventCreate = (async (props: EventCreateProps) => {
```

Comment:
nitpick(non-blocking): I would call anything handling incoming (sync) events to start with `handle` for clarity, i.e. : `handleEventCreated`

Reply:
Its not an handler, but an action. Dots are not yet supported in action names, but you can think of it as `event.create()`